greedy_algorithms/knapsack.cpp

```diff
@@ -1,78 +1,117 @@
-#include <iostream>
-using namespace std;
+/******************************************************************************
+ * @file
+ * @brief Implementation of the [Knapsack
+ * Algorithm](https://en.wikipedia.org/wiki/Knapsack_problem)
+ * @details
+ * The Knapsack Problem is a classic optimization problem in computer science
+ * and combinatorial optimization. It's about selecting a combination of items
+ * with the highest value while staying within a limited capacity (weight or
+ * volume).
+ *
+ * ## Implementation
+ *
+ * ### Step 1
+ * It initializes a 1-dimensional array dp to store the maximum profit
+ * achievable for different capacities (from 0 to the total capacity).
+ *
+ * ### Step 2
+ * It iterates over each item in the arr vector and for each item, and for each
+ * possible capacity value, it checks whether adding the current item would
+ * increase the profit. If adding the item increases the profit, it updates the
+ * dp array with the new maximum profit achievable for that capacity.
+ *
+ * ### Step 3
+ * Finally, it returns the maximum profit achievable for the given capacity.
+ *
+ * The time complexity of the knapsack algorithm is typically O(nW), where n is
+ * the number of items and W is the capacity of the knapsack. This makes it an
+ * efficient solution for moderately sized problem instances.
+ *
+ * @author [Lajat Manekar](https://github.com/Lazeeez)
+ *
+ *******************************************************************************/
 
+#include <algorithm>  // for std::max and std::fill
+#include <cassert>    // for assert
+#include <iostream>   // for I/O operations
+#include <vector>     // for std::vector
+
+// Namespace for greedy algorithms related to the knapsack problem
+namespace greedy_algorithms {
+namespace knapsack {
```

Comment:
namespaces also requires documentation 
/**
 * @namespace <name of namespace>
 * @brief what namespace does
 */